docs/documentation/custom-providers.md

```diff
@@ -31,10 +31,21 @@ Create a custom provider of data:
         public String nextInsectName() {
             return INSECT_NAMES[faker.random().nextInt(INSECT_NAMES.length)];
         }
-    }
-    ```
 
+        public String weightedInsectName() {
+            List<Map<String, Object>> insects = List.of(
+                Map.of("value", "Driver ant", "weight", 6.0),
+                Map.of("value", "Fire ant", "weight", 3.0),
+                Map.of("value", "Harvester ant", "weight", 1.0)
+            );
```

Comment:
It would be better to follow same way as existing example and place all the example data in constant like above rather than generating it for each method invocation